synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -125,28 +109,43 @@ def __init__(self, hs: "HomeServer"):
         # Whether to support MSC3772 is supported.
         self._relations_match_enabled = self.hs.config.experimental.msc3772_enabled
 
+    async def get_rules(
+        self, room_id: str, event: EventBase
+    ) -> Dict[str, List[Dict[str, dict]]]:
+        """Given an event return the rules for all users who are
+        currently in the room.
+        """
+
+        local_users = await self.store.get_local_users_in_room(room_id)
+
+        if event.type == EventTypes.Member and event.membership == Membership.JOIN:
+            if self.hs.is_mine_id(event.state_key):
+                local_users = list(local_users)
+                local_users.append(event.state_key)
+
+        ret_rules_by_user = await self.store.bulk_get_push_rules(local_users)
+
+        logger.debug("Users in room: %s", local_users)
+
+        if logger.isEnabledFor(logging.DEBUG):
+            logger.debug(
+                "Returning push rules for %r %r",
+                room_id,
+                ret_rules_by_user.keys(),
+            )
+        return ret_rules_by_user
+
     async def _get_rules_for_event(
-        self, event: EventBase, context: EventContext
+        self,
+        event: EventBase,
     ) -> Dict[str, List[Dict[str, Any]]]:
         """This gets the rules for all users in the room at the time of the event,
         as well as the push rules for the invitee if the event is an invite.
 
         Returns:
             dict of user_id -> push_rules
         """
-        room_id = event.room_id
-
-        rules_for_room_data = self._get_rules_for_room(room_id)
-        rules_for_room = RulesForRoom(
-            hs=self.hs,
-            room_id=room_id,
-            rules_for_room_cache=self._get_rules_for_room.cache,
-            room_push_rule_cache_metrics=self.room_push_rule_cache_metrics,
-            linearizer=self._rules_linearizer,
-            cached_data=rules_for_room_data,
-        )
-
-        rules_by_user = await rules_for_room.get_rules(event, context)
+        rules_by_user = await self.get_rules(event.room_id, event)
```

Comment:
May as well move the logic below above, so that we make one DB larger call rather than two mediumer ones.